packages/common/src/hardhat/tasks/finder.ts

```diff
@@ -32,39 +30,27 @@ task("setup-finder", "Points Finder to DVM system contracts")
       bridge,
       identifierwhitelist,
       mockoracle,
+      oraclespoke,
       addresswhitelist,
       financialcontractsadmin,
       store,
       optimisticoracle,
       sinkoracle,
-      prod,
-      test,
     } = taskArguments;
 
-    assert(!(sinkoracle && mockoracle), "Cannot set both SinkOracle and MockOracle to Oracle in Finder");
-
     // Determine based on task inputs which contracts to set in finder
     const contractsToSet = [];
-    if (registry || prod || test) contractsToSet.push("Registry");
-    if (generichandler || prod || test) contractsToSet.push("GenericHandler");
-    if (bridge || prod || test) contractsToSet.push("Bridge");
-    if (identifierwhitelist || prod || test) contractsToSet.push("IdentifierWhitelist");
-    if (addresswhitelist || prod || test) contractsToSet.push("AddressWhitelist");
-    if (financialcontractsadmin || prod || test) contractsToSet.push("FinancialContractsAdmin");
-    if (optimisticoracle || prod || test) contractsToSet.push("OptimisticOracle");
-    if (store || prod || test) contractsToSet.push("Store");
-
-    // TEST ONLY:
-    if (mockoracle || test) contractsToSet.push("MockOracleAncillary");
-
-    // MUST BE SPECIFICALLY SET:
-    // e.g. `yarn hardhat setup-finder --prod --sinkoracle`
+    if (registry) contractsToSet.push("Registry");
+    if (generichandler) contractsToSet.push("GenericHandler");
+    if (bridge) contractsToSet.push("Bridge");
+    if (identifierwhitelist) contractsToSet.push("IdentifierWhitelist");
+    if (addresswhitelist) contractsToSet.push("AddressWhitelist");
+    if (financialcontractsadmin) contractsToSet.push("FinancialContractsAdmin");
+    if (optimisticoracle) contractsToSet.push("OptimisticOracle");
+    if (store) contractsToSet.push("Store");
+    if (mockoracle) contractsToSet.push("MockOracleAncillary");
     if (sinkoracle) contractsToSet.push("SinkOracle");
-
-    assert(
-      !(contractsToSet.includes("SinkOracle") && contractsToSet.includes("MockOracleAncillary")),
-      "Cannot set both SinkOracle and MockOracle to Oracle in Finder"
-    );
+    if (oraclespoke) contractsToSet.push("OracleSpoke");
```

Comment:
@mrice32 FYI I removed the "OracleSpoke": "Oracle" mapping in `Constants.sol` because the `Polygon_ChildMessenger` calls `finder.getImplementationAddress("OracleSpoke")`. Instead, I changed this script to just map `OracleSpoke` to BOTH `Oracle` and the `OracleSpoke` names in the Finder.